### ISSUE TITLE
Android/ARM: opt-level = 3 produces "LLVM ERROR: Constant Island pass failed to converge" on GNU/Linux

### ISSUE THREAD

**Issue by alopatindev** [rust-lang/rust#32986]
Building `servo` (7bd2381) for Android (`./mach build --verbose --release --android`) fails when compiling `servo/.cargo/registry/src/github.com-88ac128001ac3a9a/cssparser-0.5.5/src/lib.rs`

Environment: GNU/Linux, android-ndk-11c

```
$ ./mach rustc --version --verbose
rustc 1.9.0-nightly (241a9d0dd 2016-04-05)
binary: rustc
commit-hash: 241a9d0ddf99fd40d273c615e9b1e8ce6052d94a
commit-date: 2016-04-05
host: x86_64-unknown-linux-gnu
release: 1.9.0-nightly
```

Building output:

```
...
LLVM ERROR: Constant Island pass failed to converge!
Build failed, waiting for other jobs to finish...
error: Could not compile `cssparser`.
Caused by:
  Process didn't exit successfully: `rustc /home/sbar/git/servo/.cargo/registry/src/github.com-88ac128001ac3a9a/cssparser-0.5.5/src/lib.rs --crate-name cssparser --crate-type lib -C opt-level=3 -C codegen-units=4 --cfg feature="serde" --cfg feature="heap_size" --cfg feature="heapsize" --cfg feature="heapsize_plugin" --cfg feature="serde_macros" --cfg feature="serde-serialization" -C metadata=a00db9153a066bec -C extra-filename=-a00db9153a066bec --out-dir /home/sbar/git/servo/target/arm-linux-androideabi/release/deps --emit=dep-info,link --target arm-linux-androideabi -C ar=arm-linux-androideabi-ar -C linker=/home/sbar/git/servo/components/servo/./fake-ld.sh -L dependency=/home/sbar/git/servo/target/arm-linux-androideabi/release/deps -L dependency=/home/sbar/git/servo/target/arm-linux-androideabi/release/deps --extern heapsize_plugin=/home/sbar/git/servo/target/release/deps/libheapsize_plugin-00846447151e4ac3.so --extern serde_macros=/home/sbar/git/servo/target/release/deps/libserde_macros-c550fbd19e352e36.so --extern serde=/home/sbar/git/servo/target/arm-linux-androideabi/release/deps/libserde-3a777b14c091f0d1.rlib --extern heapsize=/home/sbar/git/servo/target/arm-linux-androideabi/release/deps/libheapsize-6b0058e51ea0ec07.rlib --extern matches=/home/sbar/git/servo/target/arm-linux-androideabi/release/deps/libmatches-030a774745cc4f96.rlib --extern encoding=/home/sbar/git/servo/target/arm-linux-androideabi/release/deps/libencoding-8b01435d6d00d080.rlib --cap-lints allow -C link-args=-fuse-ld=gold -C target-feature=+neon` (exit code: 1)
```

This issue doesn't reproduce with `opt-level = 2`

Also this issue doesn't reproduce when building for x86_64 (`./mach build --verbose --release`) with `opt-level = 3`

**Comment by alexcrichton** [comment]
I believe this was initially reported as https://github.com/servo/servo/issues/10517 and @eddyb [started diagnosis](https://github.com/servo/servo/issues/10517#issuecomment-208312003) [to conclude](https://github.com/servo/servo/issues/10517#issuecomment-208376119) that [the fix is upstream](https://llvm.org/bugs/show_bug.cgi?id=25339).

We should be able to just [backport this](https://github.com/servo/servo/issues/10517#issuecomment-208439192) to our [current LLVM branch](https://github.com/rust-lang/llvm/tree/rust-llvm-2016-03-13), although it'd be best if we could throw a test case along with it.

**Comment by sanxiyn** [comment]
LLVM backport merged in #33226.

**Comment by alexcrichton** [comment]
It seems the backport was merged and I believe this was fixed, so closing.
